packages/contracts/contracts/factories/DAOFactory.sol

```diff
@@ -92,49 +79,119 @@ contract DAOFactory {
         // TODO: shall we add minter as well ?
         registry.register(_daoConfig.name, dao, msg.sender, address(token));
 
-        // create voting and initialize right away.
-        voting = ERC20Voting(
-            createProxy(
-                votingBase,
-                abi.encodeWithSelector(
-                    ERC20Voting.initialize.selector,
-                    dao,
-                    _gsnForwarder,
-                    _votingSettings[0],
-                    _votingSettings[1],
-                    _votingSettings[2],
-                    token
-                )
-            )
-        );
+        voting = createERC20Voting(dao, token, _votingSettings);
 
-        // Grant dao permission to change voting settings.
-        dao.grant(address(voting), address(dao), voting.MODIFY_VOTE_CONFIG());
+        setDAOPermissions(dao, address(voting));
 
-        ACLData.BulkItem[] memory items = new ACLData.BulkItem[](7);
+        emit DAOCreated(_daoConfig.name, address(token), address(voting));
+    }
+
+    function newWhitelistVotingDAO(
+        DAOConfig calldata _daoConfig,
+        uint256[3] calldata _votingSettings,
+        address[] calldata _whitelistVoters,
+        address _gsnForwarder
+    ) external returns (DAO dao, WhitelistVoting voting) {
+        dao = createDAO(_daoConfig, _gsnForwarder);
+
+        // register dao with its name and token to the registry
+        registry.register(_daoConfig.name, dao, msg.sender, address(0));
 
+        voting = createWhitelistVoting(dao, _whitelistVoters, _votingSettings);
+
+        setDAOPermissions(dao, address(voting));
+
+        emit DAOCreated(_daoConfig.name, address(0), address(voting));
+    }
+
+    function createDAO(DAOConfig calldata _daoConfig, address _gsnForwarder) internal returns (DAO dao) {
+        // create dao
+        dao = DAO(createProxy(daoBase, bytes("")));
+        // initialize dao with the ROOT_ROLE as DAOFactory
```

Comment:
Comment is unclear to me. Generally, NatSpec is missing for most of the functions in this contract.

Reply:
done